vividus-http-client/src/main/java/org/vividus/http/client/HttpClientConfig.java

```diff
@@ -75,7 +79,12 @@ public void setBaseUrl(String baseUrl)
 
     public boolean hasCredentials()
     {
-        return credentials != null && !credentials.isEmpty();
+        if (username != null && password != null)
+        {
+            return true;
+        }
+        isTrue(username == null && password == null, "The %s is missing", username == null ? "username" : "password");
```

Comment:
it should be moved to either a separate method (e.g. `validate` ?) or `HttpClientFactory`

Reply:
wanted to create a method validate and run it as init method to ensure that the state is valid, but we might forget to define this as init in case of custom HTTP client config bean, at the same time don't want to couple this object to spring, I'll move it into HttpClientFactory